### PR TITLE
Add parser and importer for Vulners.com reports API 

### PR DESCRIPTION
**Pull request by vankyver**
Adding vulners.com reports parser.

### CODE REVIEW

File: unittests/tools/test_vulners_parser.py
Comment (damiencarol) [inline unittests/tools/test_vulners_parser.py:12]: Please add more checks on findings

File: dojo/tools/vulners/parser.py
Comment (damiencarol) [inline dojo/tools/vulners/parser.py:74]: it seems that it should be better to use `unique_id_from_tool` for this data.

File: dojo/tools/vulners/api_client.py
Comment (damiencarol) [inline dojo/tools/vulners/api_client.py:1]: It seems that you implemented from scratch the client, any reason to not use the Pypi official package https://pypi.org/project/vulners/#description ?

**Review by damiencarol — CHANGES_REQUESTED** [review]

File: dojo/tools/vulners/api_client.py
Comment (vankyver) [reply dojo/tools/vulners/api_client.py:1]: Just to do not include one more library to project

Comment (damiencarol) [reply dojo/tools/vulners/api_client.py:1]: it's better to use this package honestly. somebody will need to maintain this part. it's better to let vulners do it itself (this package is promoted through their doc), also the package is small (in kb)

Comment (vankyver) [reply dojo/tools/vulners/api_client.py:1]: Right now vulners SDK does not support using of another API address instead of https://vulners.com
Few days ago was release of https://github.com/vulnersCom/vulners-proxy, so I can tell DD to use proxy instead of main API
As soon as SDK support proxy, I'll change raw requests to vulners SDK

File: dojo/tools/vulners/parser.py
Comment (vankyver) [reply dojo/tools/vulners/parser.py]: I kept **vulnerability_id_from_tool** and added vulners.com to **VULNERABILITY_URLS**, because I believe it's exactly public vulnerability ID like [RHSA-2018:2285](https://access.redhat.com/errata/RHSA-2018:2285) or [CESA-2021:0348](https://lists.centos.org/pipermail/centos-announce/2021-February/060805.html) and not just internal identifier

**Comment by vankyver** [comment]
@damiencarol Looks like all required fixes done. Can we finalise review?

File: dojo/tools/vulners/parser.py
Comment (damiencarol) [reply dojo/tools/vulners/parser.py]: ok. thanks

File: dojo/tools/vulners/api_client.py
Comment (damiencarol) [reply dojo/tools/vulners/api_client.py:1]: Made a review of the official SDK and it support proxies.

https://github.com/vulnersCom/api/blob/f56dab4ffcdb400da58b680d7feb89cae3550984/vulners/api.py#L97-L105

image

**Comment by damiencarol** [comment]
@vankyver :

Made a review of the official SDK and it support proxies.

https://github.com/vulnersCom/api/blob/f56dab4ffcdb400da58b680d7feb89cae3550984/vulners/api.py#L97-L105

image

**Comment by vankyver** [comment]
> @vankyver :
> 
> Made a review of the official SDK and it support proxies.
> 
> https://github.com/vulnersCom/api/blob/f56dab4ffcdb400da58b680d7feb89cae3550984/vulners/api.py#L97-L105
> 
> image

@damiencarol I've added fixes with including Vulners SDK

File: dojo/tools/vulners/api_client.py
Comment (vankyver) [reply dojo/tools/vulners/api_client.py]: @damiencarol, issues fixed. Will you have a chance to check it out?

**Review by Maffooch — APPROVED** [review]

**Review by devGregA — APPROVED** [review]

**Comment by mtesauro** [comment]
@damiencarol Wanted to give you a chance to look at this before it gets merged since you had comments above.

**Comment by devGregA** [comment]
@vankyver will merge after the release.

**Comment by damiencarol** [comment]
@vankyver will review this WE. saw a couple of things to slightly modify

File: dojo/tools/vulners/api_client.py
Comment (damiencarol) [inline dojo/tools/vulners/api_client.py:24]: why this? forcing to false make creating a connection for each call to the API

Comment (damiencarol) [reply dojo/tools/vulners/api_client.py:24]: also `server_url` already have a default value in the lib, why forcing it?

(from the 2.0.4)
```Python
def __init__(
        self, api_key, proxies=None, persistent=True, server_url="https://vulners.com"
    ):
        """
        Create VScanner API object.

        :param api_key:
            string with Vulners API key. You can obtain one from the https://vulners.com
        :param proxies:
            dict for proxy supporting. Example: {"https": "myproxy.com:3128"}
        """
```

Also, regarding our discussion, I was expecting thta you needed the support for proxies.
Something like:
```
    def get_client(self):
        return vulners.VulnersApi(
            api_key=self.api_key,
            proxies=self.proxies,  # ex: {"https": "myproxy.com:3128"}
```

File: dojo/settings/settings.dist.py
Comment (damiencarol) [inline dojo/settings/settings.dist.py:1574]:
```suggestion
```
Not needed, see my other comments

File: dojo/tools/vulners/api_client.py
Comment (damiencarol) [inline dojo/tools/vulners/api_client.py:9]: Not needed, see my other comments.

**Comment by Maffooch** [comment]
@vankyver please see feedback from Damien

**Review by manuel-sommer — CHANGES_REQUESTED** [review]
- Please add also the vulners.com parser to the docs: https://github.com/DefectDojo/django-DefectDojo/blob/master/docs/content/en/integrations/parsers.md

File: dojo/tools/vulners/api_client.py
Comment (damiencarol) [reply dojo/tools/vulners/api_client.py]: taking a look.

**Comment by manuel-sommer** [comment]
As @vankyver did not reply over a month, would it be an option to make another PR including these changes plus the requested changes from @damiencarol and close this PR?

**Comment by Maffooch** [comment]
@manuel-sommer that is cool with me!

**Comment by damiencarol** [comment]
@manuel-sommer , @Maffooch : if @vankyver is ok, I can "finish" the PR.

**Comment by Maffooch** [comment]
@damiencarol either way works! This PR has two approvals so you could change and merge at your timeline

**Comment by vankyver** [comment]
Hi, sorry, hard month, going to come back with changes asap.

File: dojo/tools/vulners/api_client.py
Comment (vankyver) [reply dojo/tools/vulners/api_client.py:24]: > This is better but the current PR doesn't support proxies. is it still one of your goal?

Proxy in this PR is not exactly a default http proxy, it's name of vulners sub-project https://github.com/vulnersCom/vulners-proxy
so server_url can be the url of vulners-proxy

But proxy in ```vulners.VulnersApi(
            api_key=self.api_key,
            proxies=self.proxies,``` is a normal http-proxy

File: dojo/settings/settings.dist.py
Comment (vankyver) [reply dojo/settings/settings.dist.py:1586]: Sorry, didn't find other comments regarding this feature. I used this to highlight links that follows to CVE and other advisories located at vulners.com.

File: dojo/tools/vulners/api_client.py
Comment (vankyver) [reply dojo/tools/vulners/api_client.py]: removed

**Review by manuel-sommer — APPROVED** [review]
thank you for adding the parser to the documentation

**Comment by mtesauro** [comment]
@damiencarol Was looking at this and was about to merge since it has 2 approvals but couldn't tell from the thread if there was work you still wanted to do.

So, I'll wait to give you a chance to chime it or address the changes you requested. :+1:

**Comment by vankyver** [comment]
@damiencarol From my point of view, everything is ready, so I hope that now we can complete the pull request.

**Review by damiencarol — APPROVED** [review]

**Comment by damiencarol** [comment]
Great work @vankyver

**Pull request merged**
